[Flaking Test] Windows unit test failing around dra plugins

### Which jobs are flaking?

https://testgrid.k8s.io/sig-windows-signal#windows-unit-master

### Which tests are flaking?

```
TestConnectionHandling/no-wipe-on-reconnect
```

### Since when has it been flaking?

Flaky for a while

### Testgrid link

https://testgrid.k8s.io/sig-windows-signal#windows-unit-master

### Reason for failure (if possible)

```
k8s.io/kubernetes/pkg/kubelet/cm/dra: plugin expand_less	13s
{Failed  === RUN   TestConnectionHandling/no-wipe-on-reconnect
=== PAUSE TestConnectionHandling/no-wipe-on-reconnect
=== CONT  TestConnectionHandling/no-wipe-on-reconnect
    registration_test.go:348: 
        	Error Trace:	C:/kubernetes/pkg/kubelet/cm/dra/plugin/registration_test.go:348
        	Error:      	Received unexpected error:
        	            	listen unix C:\Users\azureuser\AppData\Local\Temp\TestConnectionHandlingno-wipe-on-reconnect1379670676\001/dra.sock: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
        	Test:       	TestConnectionHandling/no-wipe-on-reconnect
--- FAIL: TestConnectionHandling/no-wipe-on-reconnect (0.02s)

=== RUN   TestConnectionHandling
=== PAUSE TestConnectionHandling
=== CONT  TestConnectionHandling
--- FAIL: TestConnectionHandling (0.00s)
```

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig windows node
/wg device-management

--- a/pkg/kubelet/cm/dra/plugin/dra_plugin_test.go
+++ b/pkg/kubelet/cm/dra/plugin/dra_plugin_test.go
@@ -84,13 +84,10 @@ type tearDown func()
 
 func setupFakeGRPCServer(service, addr string) (tearDown, error) {
 	ctx, cancel := context.WithCancel(context.Background())
-	teardown := func() {
-		cancel()
-	}
 
 	listener, err := net.Listen("unix", addr)
 	if err != nil {
-		teardown()
+		cancel()
 		return nil, err
 	}
 
@@ -110,19 +107,30 @@ func setupFakeGRPCServer(service, addr string) (tearDown, error) {
 			drapbv1beta1.RegisterDRAPluginServer(s, drapbv1beta1.V1ServerWrapper{DRAPluginServer: fakeGRPCServer})
 			drahealthv1alpha1.RegisterDRAResourceHealthServer(s, fakeGRPCServer)
 		} else {
+			cancel()
 			return nil, err
 		}
 	}
 
+	var wg sync.WaitGroup
+	wg.Add(1)
+
 	go func() {
-		go func() {
-			if err := s.Serve(listener); err != nil && !errors.Is(err, grpc.ErrServerStopped) {
-				panic(err)
-			}
-		}()
+		defer wg.Done()
+		if err := s.Serve(listener); err != nil && !errors.Is(err, grpc.ErrServerStopped) {
+			panic(err)
+		}
+	}()
+
+	go func() {
 		<-ctx.Done()
 		s.GracefulStop()
 	}()
+
+	teardown := func() {
+		cancel()
+		wg.Wait() // wait for Serve() to finish
+	}
 
 	return teardown, nil
 }

